vhdbuilder/packer/test/windows-files-check.ps1

```diff
@@ -179,4 +217,162 @@ function Test-ValidateSinglePackageSignature {
+        New-Item -ItemType Directory $dir -Force | Out-Null
+    }
+
+    $excludeHashComparisionListInAzureChinaCloud = @(
+        "calico-windows",
+        "azure-vnet-cni-singletenancy-windows-amd64",
+        "azure-vnet-cni-singletenancy-swift-windows-amd64",
+        "azure-vnet-cni-singletenancy-overlay-windows-amd64",
+        # We need upstream's help to republish this package. Before that, it does not impact functionality and 1.26 is only in public preview
+        # so we can ignore the different hash values.
+        "v1.26.0-1int.zip"
+    )
+
+    foreach ($URL in $map[$dir]) {
+        $fileName = [IO.Path]::GetFileName($URL)
+        $dest = [IO.Path]::Combine($dir, $fileName)
+
+        DownloadFileWithRetry -URL $URL -Dest $dest -redactUrl
+        $globalFileSize = (Get-Item $dest).length
+        
+        $isIgnore=$False
+        foreach($excludePackage in $excludeHashComparisionListInAzureChinaCloud) {
+            if ($URL.Contains($excludePackage)) {
+                $isIgnore=$true
+                break
+            }
+        }
+        if ($isIgnore) {
+            continue
+        }
+
+        if ($URL.StartsWith("https://acs-mirror.azureedge.net/")) {
+            $mcURL = $URL.replace("https://acs-mirror.azureedge.net/", "https://kubernetesartifacts.blob.core.chinacloudapi.cn/")
+
+            $mooncakeFileSize = (Invoke-WebRequest $mcURL -UseBasicParsing -Method Head).Headers.'Content-Length'
+
+            if ($globalFileSize -ne $mooncakeFileSize) {
+                $MisMatchFile[$URL]=$mcURL
+            }
+        }
+    }
+}
+
+# Compare Files on Global and MoonCake
+function Test-CompareFiles {
+    foreach ($dir in $map.Keys) {
+        Test-CompareSingleDir $dir
+    }
+
+    if ($MisMatchFile.Count -ne 0) {
+        $MisMatchFile = (echo $MisMatchFile | ConvertTo-Json -Compress)
+        Write-Error "The following files have different sizes on global and mooncake: $MisMatchFile"
+    }
+}
+
+function Retry-Command {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Position=0, Mandatory=$true)]
+        [scriptblock]$ScriptBlock,
+
+        [Parameter(Position=1, Mandatory=$true)]
+        [string]$ErrorMessage,
+
+        [Parameter(Position=2, Mandatory=$false)]
+        [int]$Maximum = 5,
+
+        [Parameter(Position=3, Mandatory=$false)]
+        [int]$Delay = 10
+    )
+
+    Begin {
+        $cnt = 0
+    }
+
+    Process {
+        do {
+            $cnt++
+            try {
+                $ScriptBlock.Invoke()
+                if ($LASTEXITCODE) {
+                    throw "Retry $cnt : $ErrorMessage"
+                }
+                return
+            } catch {
+                Write-Error $_.Exception.InnerException.Message -ErrorAction Continue
+                if ($_.Exception.InnerException.Message.Contains("There is not enough space on the disk. (0x70)")) {
+                    Write-Error "Exit retry since there is not enough space on the disk"
+                    break
```

Comment:
Throw the error. Otherwise it seems like that ErrorMessage does not contain "Exit retry since there is not enough space on the disk"